File: 2-AD-HOC/1026-Carrega_ou_nao_Carrega-5/1026.c
#include <stdio.h>
#include <math.h>

void bin(double n, double array[33]) {
    int indic=32;
    while (n > 0) {
        double resto = fmod(n, 2);
        n = floor(n/2);
        array[indic]=resto;
        indic--;
    }

}


int main() {
    while(1){
        double n1, n2, cv, resultado;
        resultado=0;
        if (scanf("%lf", &n1) == -1){
            break;
        }
        
        if (scanf("%lf", &n2) == -1){
            break;
        }
        
        double array1[33]={0};
        bin(n1, array1);
        double array2[33]={0};
        bin(n2, array2);
        for (int i =0; i<33; i++){
            cv=array1[i]+array2[i];
            if (cv ==2){
                cv=0;
            }
            resultado+=cv*pow(2, 33-(i+1));   
           
        }
         printf("%.0f\n", resultado);

        //10101010101010101010101010101010
        // 1010101010101010101010101010101
        
        }
    return 0;
}

File: Beecrowd/2-AD-HOC/1026-Carrega_ou_nao_Carrega-5/1026.py
import sys
import math

def bin_representation(n, length=33):
    array = [0] * length
    indic = length - 1
    while n > 0:
        resto = n % 2
        n = n // 2
        array[indic] = resto
        indic -= 1
    return array

input = sys.stdin.read().strip().split()
index = 0

while index < len(input):
    if index + 1 >= len(input):
        break

    n1 = float(input[index])
    n2 = float(input[index + 1])
    index += 2

    array1 = bin_representation(int(n1))
    array2 = bin_representation(int(n2))

    resultado = 0
    for i in range(33):
        cv = array1[i] + array2[i]
        if cv == 2:
            cv = 0
        resultado += cv * (2 ** (32 - i))

    print(int(resultado))


